### PR TITLE
Use linage in KMeansMiniBatch

### PR DESCRIPTION
**Pull request by LiuYuHui**
change manual loops to linalg. #2747

### CODE REVIEW

File: src/shogun/clustering/KMeansMiniBatch.cpp [hunk 1]
```diff
@@ -71,18 +72,7 @@ void KMeansMiniBatch::minibatch_KMeans()
...
-			for (int32_t p=1; p<k; p++)
-			{
-				if (dists[p]<min)
-				{
-					imin=p;
-					min=dists[p];
-				}
-			}
-			ncent[j]=imin;
+			ncent[j] = Math::arg_max(dists.vector, 1, dists.vlen);
```
Comment (gf712) [inline src/shogun/clustering/KMeansMiniBatch.cpp]: I thought this was meant to find argmin?

Comment (LiuYuHui) [reply src/shogun/clustering/KMeansMiniBatch.cpp]: oh, yes, I will change it to argmin, thanks you point it.

Comment (vigsterkr) [reply src/shogun/clustering/KMeansMiniBatch.cpp]: huh if this has not been caught by CI means that we dont have a test that implicitly tests stuff of this part of the code.... @LiuYuHui might be good to add some unit test that covers this :)

Comment (vigsterkr) [reply src/shogun/clustering/KMeansMiniBatch.cpp]: plz avoid testing private functions of KMeansMiniBatch.... just public interface

Comment (LiuYuHui) [reply src/shogun/clustering/KMeansMiniBatch.cpp]: ok, I am very happy to do it!

Comment (vigsterkr) [reply src/shogun/clustering/KMeansMiniBatch.cpp]: great thnx! just add it to this PR

Comment (LiuYuHui) [reply src/shogun/clustering/KMeansMiniBatch.cpp]: hey @vigsterkr , does it look fine to you?

File: tests/unit/clustering/kmeans_unittest.cc [hunk 2]
```diff
@@ -168,6 +168,35 @@ TEST(KMeans, minibatch_training_test)
...
+
+	clustering->put<int32_t>("max_iter", 100000);
+	clustering->put<int32_t>("batch_size", 12);
+	clustering->train(features);
+	auto learnt_centers_matrix = clustering->get_cluster_centers();
+
+	EXPECT_NEAR(2.1666612499593461, learnt_centers_matrix(0, 0), 0.0001);
+	EXPECT_NEAR(2.8888888888888888, learnt_centers_matrix(0, 1), 0.0001);
+	EXPECT_NEAR(2.4999954166322538, learnt_centers_matrix(1, 0), 0.0001);
+	EXPECT_NEAR(3.1111111111111116, learnt_centers_matrix(1, 1), 0.0001);
```
Comment (gf712) [inline tests/unit/clustering/kmeans_unittest.cc:192]: @LiuYuHui how did you test these?

Comment (LiuYuHui) [reply tests/unit/clustering/kmeans_unittest.cc:192]: I assume the previous implementation is right, this is the results of the previous implementation :)

Comment (gf712) [reply tests/unit/clustering/kmeans_unittest.cc:192]: Hmmm I guess the values make sense...

File: tests/unit/clustering/kmeans_unittest.cc [hunk 3]
```diff
@@ -168,6 +168,35 @@ TEST(KMeans, minibatch_training_test)
...
+
+	SGMatrix<float64_t> X{{2, 1}, {4, 1}, {0, 1}, {2, 4}, {0, 4}, {4, 4},
+	                      {5, 4}, {1, 0}, {2, 2}, {2, 3}, {5, 5}, {-1, 1}};
+
+	SGMatrix<float64_t> initial_centers(2, 2);
+	initial_centers(0, 0) = 0;
+	initial_centers(1, 0) = 0;
+
+	initial_centers(0, 1) = 1;
+	initial_centers(1, 1) = 1;
```
Comment (gf712) [inline tests/unit/clustering/kmeans_unittest.cc]: you could write this a one liner `SGMatrix<float64_t> initial_centers{{0, 1}, {0, 1}}` no?

Comment (LiuYuHui) [reply tests/unit/clustering/kmeans_unittest.cc]: done

**Pull request merged**